src/main/python/ccrf/prepare_test_data.py

```diff
@@ -10,61 +10,69 @@
 
 from utils import TfidfTgzReader
 
-topic_list = [
-    '321', '336', '341',
-    '347', '350', '362',
-    '363', '367', '375', '378', '393',
-    '397', '400', '408', '414',
-    '422', '426', '427', '433',
-    '439', '442', '445', '626', '646',
-    '690'
-]
+# Global variables, which we're later going to refactor into an external config file.
+
+topics = ['321', '336', '341', '347', '350', '362', '363', '367', '375', '378',
+          '393', '397', '400', '408', '414', '422', '426', '427', '433', '439',
+          '442', '445', '626', '646', '690']
+
+config = {
+    'target': {'name': 'core18',
+               'run': 'run.core18.bm25+rm3.topics.core18.txt',
+               'index': 'lucene-index.core18.pos+docvectors+rawdocs'}
+}
+
+working_directory = 'ccrf'
 
-def read_vocab(path):
-    """
 
-    """
+def dump_docvectors():
+    run = config['target']['run']
+    source_name = config['target']['name']
+    index = config['target']['index']
+    docids_file = os.path.join(working_directory, f'docids.{source_name}')
+    logging.info(f'Extracting docvectors for {source_name} with Anserini')
+    cmd = f'cut -f 3 -d " " {run} | sort | uniq > {docids_file}'
+    os.system(cmd)
+    cmd = f'target/appassembler/bin/IndexUtils -index {index} -dumpDocVectors {docids_file} -docVectorWeight TF_IDF'
+    os.system(cmd)
+
+
+def read_vocab(path):
     logging.info("loading vocabulary dictionary...")
     with open(path, 'rb') as f:
         vocab = pickle.load(f)
 
     return vocab
 
-def build_docid_idx_dict(rank_file):
-    """
-
-    """
+def build_docid_idx_dict():
     logging.info('building docid idx dict...')
     cur_idx = 0
     docid_idx_dict = {}
-    with open(rank_file, 'r') as f:
+    with open(config['target']['run'], 'r') as f:
         for line in f:
             topic, _, docid, _, _, _ = line.split(' ')
-            if topic in topic_list and docid not in docid_idx_dict:
+            if topic in topics and docid not in docid_idx_dict:
```

Comment:
Although the list is quite small, considering the judgement will be called for every line, I suggest use a set instead of list.

experiments
```
%%timeit
for topic in topics:
  if topic in topics:
    pass
>>> 100000 loops, best of 3: 14.4 µs per loop

%%timeit
for topic in topics:
  if topic in topics_set:
    pass
>>> 100000 loops, best of 3: 2.41 µs per loop
```